lxd/db/storage_pools.go

```diff
@@ -273,31 +277,27 @@ func (c *ClusterTx) CreatePendingStoragePool(node, name, driver string, conf map
 		state  StoragePoolState
 	}{}
 
-	var errConsistency error
-	dest := func(i int) []any {
-		// Ensure that there is at most one pool with the given name.
-		if i != 0 {
-			errConsistency = fmt.Errorf("more than one pool exists with the given name")
-		}
-
-		return []any{&pool.id, &pool.driver, &pool.state}
-	}
-
 	stmt, err := c.tx.Prepare("SELECT id, driver, state FROM storage_pools WHERE name=?")
 	if err != nil {
 		return err
 	}
 
 	defer func() { _ = stmt.Close() }()
-	err = query.SelectObjects(stmt, dest, name)
+	count := 0
+	err = query.SelectObjects(stmt, func(scan func(dest ...any) error) error {
```

Comment:
Why don't we use query.QueryScan here instead of preparing a one-time query above?